### PR TITLE
DCS-1370 adding new endpoint for prisoner returning from court

### PR DESCRIPTION
**Pull request by michalnawrockidj**

### CODE REVIEW

File: NAMING_CONVENCTION.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# hmpps-welcome-people-into-prison-api Naming Convention
```
Comment (andrewrlee) [inline NAMING_CONVENCTION.md:1]: typo in file name

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsService.kt [hunk 2]
```diff
@@ -57,34 +59,46 @@ class ArrivalsService(
 
     val arrival = getArrival(moveId)
 
+    return if (!arrival.isCurrentPrisoner)
+      when (arrival.prisonNumber) {
+        null -> createAndAdmitOffender(confirmArrivalDetail, moveId)
+        else -> admitOffender(confirmArrivalDetail, moveId, arrival.prisonNumber)
+      } else
+      throw IllegalArgumentException("Feature not available.")
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsService.kt:67]: can we make this more descriptive: `Do not support confirming arrival of active prisoner: ${arrival.prisonNumber}`

Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsService.kt:67]: or summat

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsService.kt [hunk 3]
```diff
@@ -57,34 +59,46 @@ class ArrivalsService(
...
       when (arrival.fromLocationType) {
-        LocationType.COURT -> transferInFromCourt(moveId, confirmArrivalDetail, arrival)
+        LocationType.COURT -> returnFromCourt(moveId, confirmCourtReturnRequest, arrival)
         else -> throw IllegalArgumentException("The arrival is known to NOMIS, has a current booking but is not from court. This scenario is not supported.")
       } else
-      when (arrival.prisonNumber) {
-        null -> createAndAdmitOffender(confirmArrivalDetail, moveId)
-        else -> admitOffender(confirmArrivalDetail, moveId, arrival.prisonNumber)
-      }
+      throw IllegalArgumentException("Feature not available.")
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsService.kt:82]: can we make this more descriptive? 
`Confirming arrival of inactive prisoner: '${arrival.prisonNumber}' is not supported`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonService.kt [hunk 4]
```diff
@@ -78,16 +79,16 @@ class PrisonService(
...
     return prisonApiClient.courtTransferIn(
       arrival.prisonNumber!!,
-      with(confirmArrivalDetail) {
+      with(confirmCourtReturnRequest) {
         CourtTransferIn(
           prisonId!!,
-          movementReasonCode,
-          commentText,
-          bookingInTime
+          null,
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonService.kt:89]: default these to null in the `CourtTransferIn` so don't have to pass them in here?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonServiceTest.kt [hunk 5]
```diff
@@ -159,15 +152,15 @@ class PrisonServiceTest {
...
       arrival.prisonNumber!!,
       CourtTransferIn(
-        confirmArrivalDetail.prisonId!!,
-        confirmArrivalDetail.movementReasonCode,
-        confirmArrivalDetail.commentText,
-        confirmArrivalDetail.bookingInTime
+        confirmCourtReturnRequest.prisonId!!,
+        null,
+        null,
+        null
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonServiceTest.kt:163]: once these are removed or defaulted then can remove these nulls

File: NAMING_CONVENCTION.md (hunk 1 above)
Comment (michalnawrockidj) [reply NAMING_CONVENCTION.md:1]: done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsService.kt (hunk 2 above)
Comment (michalnawrockidj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsService.kt:67]: done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsService.kt (hunk 3 above)
Comment (michalnawrockidj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsService.kt:82]: done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonService.kt (hunk 4 above)
Comment (michalnawrockidj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonService.kt:89]: done

File: src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonServiceTest.kt (hunk 5 above)
Comment (michalnawrockidj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonServiceTest.kt:163]: done

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**